arm-apimanagement/2015-09-15/swagger/service.json

```diff
@@ -0,0 +1,1288 @@
+        ],
+        "operationId": "ApiManagement_BackupApiManagementServiceToCloudBlob",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/APIManagementServiceNameParameter"
+          },
+          {
+            "name": "destination",
+            "in": "body",
+            "schema": {
+              "$ref": "#/definitions/BackupRestoreAPIManagementServiceContract"
+            },
+            "x-nullable": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "type": "file"
+            },
+            "x-nullable": true
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/restore": {
+      "post": {
+        "tags": [
+          "ApiManagement"
+        ],
+        "operationId": "ApiManagement_RestoreApiManagementServiceFromCloudBlob",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/APIManagementServiceNameParameter"
+          },
+          {
+            "name": "source",
+            "in": "body",
+            "schema": {
+              "$ref": "#/definitions/BackupRestoreAPIManagementServiceContract"
+            },
+            "x-nullable": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "type": "file"
+            },
+            "x-nullable": true
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "UserSubscription": {
+      "type": "object",
+      "x-typeName": "UserSubscription",
+      "required": [
+        "RegistrationDate",
+        "State",
+        "ForceDeleteServices"
+      ],
+      "properties": {
+        "SubscriptionId": {
+          "type": "string"
+        },
+        "RegistrationDate": {
+          "type": "string",
+          "format": "date-time"
+        },
+        "State": {
+          "$ref": "#/definitions/UserSubscriptionState"
+        },
+        "Properties": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "ForceDeleteServices": {
+          "type": "boolean"
+        }
+      }
+    },
+    "UserSubscriptionState": {
+      "type": "string",
+      "x-typeName": "UserSubscriptionState",
+      "x-enumNames": [
```

Comment:
We have an extesnion named "x-ms-enum". Please find all of the extensions supported by Autorest over here: https://github.com/Azure/autorest/blob/master/Documentation/swagger-extensions.md